app/parse.py

```diff
@@ -15,8 +38,77 @@ class Product:
     num_of_reviews: int
 
 
+PRODUCT_FIELDS = [field.name for field in fields(Product)]
+
+_driver: WebDriver | None = None
+
+
+def get_driver() -> WebDriver:
+    return _driver
+
+
+def set_driver(new_driver: WebDriver) -> None:
+    global _driver
+    _driver = new_driver
+
+
+def parse_single_product(product_soup: BeautifulSoup) -> Product:
+    return Product(
+        title=product_soup.select_one(".title")["title"],
+        description=product_soup.select_one(
+            ".description").text.replace("\xa0", " "),
+        price=float(product_soup.select_one(".price").text.replace("$", "")),
+        rating=len(product_soup.select(".glyphicon-star")),
+        num_of_reviews=int(
+            product_soup.select_one(".ratings > p.pull-right").text.split()[0]
+        ),
+    )
+
+
+def get_single_page_products(url: str) -> list[Product]:
+    driver = get_driver()
+    driver.get(url)
+
+    try:
+        accept_cookies_button = driver.find_element(
+            By.CLASS_NAME, "acceptCookies"
+        )
+        if accept_cookies_button:
+            accept_cookies_button.click()
+    except Exception:
+        pass
```

Comment:
specify a specific exception class
instead of "pass" specify something specific like printing information about the exception